sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -975,13 +1047,21 @@ public WriteResult expand(PCollection<T> input) {
               getTableFunction(), getFormatFunction()))
               .setCoder(KvCoder.of(TableDestinationCoder.of(), TableRowJsonCoder.of()));
 
+      // The transforms always expect to get a non-null schema function.
+      SchemaFunction schemaFunction = this.getSchemaFunction();
+      if (schemaFunction == null) {
```

Comment:
You can use MoreObjects.firstNonNull. Or you can put ConstantSchemaFunction.fromJsonSchema(null) into the builder in BigQueryIO.write().